src/sourceSystem.cpp

```diff
@@ -136,14 +143,11 @@ int SourceSystemCVODE::f(const realtype t, const sdVector& y, sdVector& ydot)
 
     updateThermo();
 
-    double a = strainFunction->a(t);
-    double dadt = strainFunction->dadt(t);
-
     // *** Calculate the time derivatives
     double scale;
     if (!quasi2d) {
         scale = 1.0;
-        dUdt = - U*U + rhou/rho*(dadt + a*a) + splitConst[kMomentum];
+        dUdt = - U*U + splitConst[kMomentum];
```

Comment:
Did you try moving the U*U term to the convection system as well? That way, it would eliminate the operator splitting error associated with this term entirely. Seems like it should work well, given that the da/dt + a^2 and U^2 terms normally balance each other out to a large extent.

Reply:
I just tried this, but it seemed to have no impact on computational time. Easy enough change to make though if you would prefer U^2 be in convectionSystem as well.